File: Client/src/entities/users/api/update-user/update-user-command.ts
import { Gender } from "../../model";

export type UpdateUserCommand = {
	userId: string;
	updatedUserData: UserData;
};

export type UserData = {
	firstName?: string | null;
	lastName?: string | null;
	email?: string | null;
	phoneNumber?: string | null;
	gender?: Gender | null;
	country?: string | null;
	city?: string | null;
	zipCode?: string | null;
	street?: string | null;
};

File: Server/Server.Application/Users/UpdateUser/UpdateUserCommand.cs
using Server.Application.Abstractions.Messaging;
using Server.Domain.Users;

namespace Server.Application.Users.UpdateUser;

public sealed record UpdateUserCommand(
    Guid UserId,
    string? FirstName,
    string? LastName,
    string? Email,
    string? PhoneNumber,
    Gender? Gender,
    string? Country,
    string? City,
    string? ZipCode,
    string? Street) : ICommand;
 
